app/src/main/java/com/github/alfabravo2013/readyforexams/MainActivity.kt

```diff
@@ -2,18 +2,43 @@ package com.github.alfabravo2013.readyforexams
 
 import android.os.Bundle
 import androidx.appcompat.app.AppCompatActivity
+import androidx.navigation.fragment.NavHostFragment
 import com.github.alfabravo2013.readyforexams.databinding.ActivityMainBinding
 
 class MainActivity : AppCompatActivity() {
     private lateinit var binding: ActivityMainBinding
+    private var onNavigateUpCallback: (() -> Unit)? = null
```

Comment:
I m a bit confused about why you need this callback here? You want to make the back navigation from Toolbar being available from each fragment?